### ISSUE TITLE
How does keras multiple losses affect multiple model branchs? 

### ISSUE THREAD

**Issue by Buckler89** [keras-team/keras#10739]
I found just two answers [here](https://stackoverflow.com/questions/49404309/how-does-keras-handle-multiple-losses).
The two answers in the link seem to be contrastive.
Looking at the code seem that the right answer is @wishcome's one. But in my opinion, the right way of doing the backpropagation in the case of the example proposed by @Fábio Perez is that the same as proposed by him.
I report an example (proposed by Fábio Perez-see link) of what I think keras should do:
```
                        +----+         
                        > C  |-->loss1 
                       /+----+         
                      /                
                     /                 
    +----+    +----+/                  
 -->| A  |--->| B  |\                  
    +----+    +----+ \                 
                      \                
                       \+----+         
                        > D  |-->loss2 
                        +----+     
```

- loss1 will affect A, B, and C.
- loss2 will affect A, B, and D.

What do you think?

**Comment by chriskoups** [comment]
I think that must go back to the Neural Network theory.
The loss (summation of the weighted losses) will update everything but the gradients will update the weights so, if one of the branches is increasing the loss its gradient will take care of it.
To sum up, loss is used in the gradient calculation of the individual weights. You may have the weight updates a bit messed up. I will suggest you to read a paper in essential NN equations.
